src/nrniv/partrans.cpp

```diff
@@ -27,10 +27,10 @@
 
 #if NRNLONGSGID
 #define sgid_t int64_t
-#define sgid_alltoallv nrnmpi_long_alltoallv
+#define sgid_alltoallv nrnmpi_long_alltoallv_sparse
 #else
 #define sgid_t int
-#define sgid_alltoallv nrnmpi_int_alltoallv
+#define sgid_alltoallv nrnmpi_int_alltoallv_sparse
```

Comment:
I guess we have to create a couple of wrapper functions that use nrn_sparse_partrans.